homeassistant/components/fireservicerota/sensor.py

```diff
@@ -0,0 +1,126 @@
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN as FIRESERVICEROTA_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up FireServiceRota sensor based on a config entry."""
+    coordinator = hass.data[FIRESERVICEROTA_DOMAIN][entry.entry_id]
+
+    async_add_entities([IncidentsSensor(coordinator, entry)])
+
+
+class IncidentsSensor(RestoreEntity, Entity):
+    """Representation of FireServiceRota incidents sensor."""
+
+    def __init__(self, coordinator, entry):
+        """Initialize."""
+        self._coordinator = coordinator
+        self._entry_id = entry.entry_id
+        self._unique_id = entry.unique_id
+
+        self._state = None
+        self._state_attributes = {}
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return "Incidents"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        if "prio" in self._state_attributes:
+            if self._state_attributes["prio"][0] == "a":
+                return "mdi:ambulance"
+
+        return "mdi:fire-truck"
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{self._unique_id}_Incidents"
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return available attributes for sensor."""
+        attr = {}
+        data = self._state_attributes
+
+        if data:
+            for value in (
+                "trigger",
+                "created_at",
+                "message_to_speech_url",
+                "prio",
+                "type",
+                "responder_mode",
+                "can_respond_until",
+            ):
+                if data.get(value):
+                    attr[value] = data[value]
+
+            if "address" in data:
+                for address_value in (
+                    "latitude",
+                    "longitude",
+                    "address_type",
+                    "formatted_address",
+                ):
+                    if address_value in data["address"]:
+                        attr[address_value] = data["address"][address_value]
+
+            return attr
+
+    async def async_added_to_hass(self) -> None:
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
+
+        state = await self.async_get_last_state()
```

Comment:
Why is this necessary as the API allows refreshing data?

Reply:
This is part of 'RestoreEntity' code to prevent 'unknown' state and attributes after restarts, is it wrong?

Reply:
It not only updates the listener data (which indeed is not needed) but it does also update the binary_sensor and switch state, which are polled.

Reply:
I am confused here. This is the sensor entity. How are the binary sensor / switch related?

Are you unable to poll the incidents?

Reply:
When a web-socket client connection was made recently before a restart it would get the last event again on the websocket channel I connect to and listen for events on. But if the time between them is longer, like a few minutes, you get nothing, having an unknown state is confusing. So that's why I did it like this, to have a clean history.

Reply:
The response switch used the event_id from the sensor state. Since a response is linked to an emergency event. So when I restore the state of the sensor, the state of the switch is also correctly fetched.

Reply:
So this is only the case when the firefighter was restarting their Home Assistant instance after the alert came in?